scripts/timelock-deploy.js

```diff
@@ -0,0 +1,118 @@
+const { network } = require("hardhat");
+const CONFIG = require("./addresses.js");
+const { verifyTimelock } = require("./timelock-verify.js");
+const { renounceRole } = require("./timelock-renounceRole.js");
+
+async function main(config) {
+  // This is just a convenience check
+  if (!config) {
+    config = CONFIG[network.name];
+  }
+
+  const [deployer] = await ethers.getSigners();
+  console.log(
+    "Deploying the contracts with the account:",
+    await deployer.getAddress()
+  );
+
+  const governance = config.governance;
+  var hatGovernanceDelay;
+  if (config.minDelay) {
+    hatGovernanceDelay = config.minDelay;
+  } else {
+    hatGovernanceDelay = 60 * 60 * 24 * 7; // 7 days
+  }
+  const hatVaultsAddress = config.hatVaultsAddress;
+  const executors = config.executors;
+
+  console.log("Account balance:", (await deployer.getBalance()).toString());
+
+  const HATTimelockController = await ethers.getContractFactory(
+    "HATTimelockController"
+  );
+  console.log(`waiting for HATTimeLockController to deploy..`);
+  var hatTimelockController;
+  if (network.name !== "hardhat") {
+    hatTimelockController = await HATTimelockController.deploy(
+      hatVaultsAddress,
+      hatGovernanceDelay, // minDelay
+      [governance], // proposers
+      executors // executors
+    );
+    await hatTimelockController.deployed();
+  } else {
+    // if network is hardhat, then we are running a test, and we use a different deploymnet method
+    const HATTimelockControllerArtifact = artifacts.require(
+      "./HATTimelockController.sol"
+    );
+    hatTimelockController = await HATTimelockControllerArtifact.new(
+      hatVaultsAddress,
+      hatGovernanceDelay, // minDelay
+      [governance], // proposers
+      executors // executors
+    );
+  }
+  console.log(
+    `HATTimeLockController is deployed at`,
+    hatTimelockController.address
+  );
+
+  // the deployer automatically gets the admin role, which we renounce (so all admin has to go through the proposer)
+  await renounceRole({
+    timelock: hatTimelockController.address,
+    deployer: deployer.address,
+  });
+
+  // We also save the contract's artifacts and address in the frontend directory
+  if (network.name !== "hardhat") {
+    await verifyTimelock(
+      config.update({ timelock: hatTimeLockController.address })
```

Comment:
This update method doesn't seem to exist. Should update like: `config["timelock"] = hatTimelockController.address;` on the line before then just pass config.

Reply:
what version of node are u using?

Reply:
anway this script is tested, so if CI says it is ok then the problem is yours :)

Reply:
Oh yeah bad node version sorry